cdap-ui/app/cdap/components/AbstractWidget/HierarchyWidget/dataHandler.ts

```diff
@@ -0,0 +1,390 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import { IStageSchema } from 'components/AbstractWidget';
+import uuidV4 from 'uuid/v4';
+
+interface IFieldType {
+  name: string;
+  type: any;
+}
+
+function enumHandler(field) {
+  return { name: field.name, type: field.type.type, children: field.type.symbols };
+}
+
+function mapHandler(field) {
+  const objectKeys = Object.keys(field.type);
+  const objectValues = Object.values(field.type);
+  return {
+    name: field.name,
+    type: field.type.type,
+    children: [
+      { name: objectKeys[1], type: objectValues[1] },
+      { name: objectKeys[2], type: objectValues[2] },
+    ],
+  };
+}
+
+function intByteHandler(field) {
+  return { name: field.name, type: field.type.logicalType, children: [] };
+}
+
+function longHandler(field) {
+  return { name: field.name, type: field.type.logicalType, children: [] };
+}
+
+function recordHandler(field) {
+  return {
+    name: field.name,
+    type: field.type.type,
+    children: field.type.fields,
+  };
+}
+
+function arrayHandler(field) {
+  return { name: field.name, type: field.type.type, children: [field.type.items] };
+}
+
+function stringHandler(field) {
+  return { name: field.name, type: field.type, children: [] };
+}
+
+// The function receives a field from inputschema and is meant to destructure and
+// return it in the format that the Autocomplete component is expecting
+function getType(field: IFieldType) {
+  let fieldFormat = {};
+
+  switch (typeof field.type) {
+    case 'object':
+      switch (field.type.type) {
+        case 'enum':
+          fieldFormat = enumHandler(field);
+          break;
+        case 'map':
+          fieldFormat = mapHandler(field);
+          break;
+        case 'bytes':
+        case 'int':
+          fieldFormat = intByteHandler(field);
+          break;
+        case 'long':
+          fieldFormat = longHandler(field);
+          break;
+        case 'record':
+          fieldFormat = recordHandler(field);
+          break;
+        case 'array':
+          fieldFormat = arrayHandler(field);
+          break;
+        default:
+          break;
+      }
+      return fieldFormat;
+    case 'string':
+      fieldFormat = stringHandler(field);
+      break;
+    default:
+      break;
+  }
+  return fieldFormat;
+}
+
+// This function receives a field from the input schema and depending on the condition returns the expected format
+function typeHandler(field) {
+  let fieldFormat = {};
+  if (Array.isArray(field.type)) {
+    const objToFormat = {
+      name: field.name,
+      type: field.type[0],
+    };
+    fieldFormat = getType(objToFormat);
+    typeHandler(objToFormat);
+  } else {
+    fieldFormat = getType(field);
+  }
+  return fieldFormat;
+}
+interface IField {
+  name: string;
+  type: string[];
```

Comment:
Is this always string array? This will be array if the field is of null type but it can be as simple as string or it can get complicated as nested types.